x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/index.tsx

```diff
@@ -526,6 +620,97 @@ const PoliciesBreadcrumb: React.FunctionComponent<{ policyName: string; policyId
   return null;
 };
 
+const UpgradeStatusCallout: React.FunctionComponent<{
+  dryRunData: UpgradePackagePolicyDryRunResponse;
+}> = ({ dryRunData }) => {
+  const [isPreviousVersionFlyoutOpen, setIsPreviousVersionFlyoutOpen] = useState<boolean>(false);
+
+  if (!dryRunData) {
+    return null;
+  }
+
+  const isReadyForUpgrade = !dryRunData[0].hasErrors;
+
+  const [currentPackagePolicy, proposedUpgradePackagePolicy] = dryRunData[0].diff || [];
+
+  const FlyoutBody = styled(EuiFlyoutBody)`
+    .euiFlyoutBody__overflowContent {
+      padding: 0;
+    }
+  `;
+
+  return (
+    <>
+      {isPreviousVersionFlyoutOpen && (
+        <EuiPortal>
+          <EuiFlyout onClose={() => setIsPreviousVersionFlyoutOpen(false)} size="l" maxWidth={640}>
+            <EuiFlyoutHeader hasBorder>
+              <EuiTitle size="m">
+                <h2 id="FleetPackagePolicyPreviousVersionFlyoutTitle">
+                  <FormattedMessage
+                    id="xpack.fleet.upgradePackagePolicy.previousVersionFlyout.title"
+                    defaultMessage="'{name}' package policy"
+                    values={{ name: currentPackagePolicy?.name }}
+                  />
+                </h2>
+              </EuiTitle>
+            </EuiFlyoutHeader>
+            <FlyoutBody>
+              <EuiCodeBlock isCopyable fontSize="m" whiteSpace="pre">
+                {JSON.stringify(currentPackagePolicy, null, 2)}
```

Comment:
I am wondering here if it will make more sense for the user to show the compiled package policy as it's sent to the agent? we never show the package policy and I think we should probably hide that complexity to the user.

Reply:
Yeah definitely. Is there a utility for compiling it into that format?

Reply:
maybe that function? https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/common/services/package_policies_to_agent_inputs.ts#L13